spot-vaults/contracts/BillBroker.sol

```diff
@@ -633,22 +645,36 @@ contract BillBroker is
     function computeMintAmtWithUSD(
         uint256 usdAmtIn,
         ReserveState memory s
-    ) public view returns (uint256) {
+    ) public view returns (uint256 mintAmt) {
         if (usdAmtIn <= 0) {
             return 0;
         }
 
         // We compute equal value of perp tokens going out.
+        uint256 totalSupply_ = totalSupply();
         uint256 valueIn = s.usdPrice.mulDiv(usdAmtIn, usdUnitAmt);
         uint256 totalReserveVal = (s.usdPrice.mulDiv(s.usdBalance, usdUnitAmt) +
             s.perpPrice.mulDiv(s.perpBalance, perpUnitAmt));
-        return
-            (totalReserveVal > 0)
-                ? valueIn.mulDiv(totalSupply(), totalReserveVal).mulDiv(
-                    ONE - fees.mintFeePerc,
-                    ONE
-                )
-                : 0;
+        if (totalReserveVal == 0 || totalSupply_ == 0) {
+            return 0;
+        }
+
+        // Compute mint amount.
+        mintAmt = valueIn.mulDiv(totalSupply_, totalReserveVal);
+
+        // Apply a combination of swap and mint fees.
+        uint256 percOfAmtInSwapped = ONE.mulDiv(
+            usdAmtIn - mintAmt.mulDiv(s.usdBalance, totalSupply_),
+            usdAmtIn
+        );
+        uint256 feeFactor = computeUSDToPerpSwapFeeFactor(
+            assetRatio(s),
+            assetRatio(_updatedReserveState(s, s.usdBalance + usdAmtIn, s.perpBalance))
+        );
+        mintAmt =
```

Comment:
Let's waive the mint fee on single sided deposits. It's good to encourage depositors who balance out the pool.